stable/rabbitmq-ha/templates/statefulset.yaml

```diff
@@ -146,6 +146,12 @@ spec:
             {{- end }}
           resources:
 {{ toYaml .Values.resources | indent 12 }}
+          {{- if .Values.securityContext.enabled }}
+          securityContext:
```

Comment:
Can we move this to pod level securityContext so all the containers have reduced permissions?

Reply:
It is possible although rabbitmq-exporter still runs with gid 0:
```
[root@ii-k8s-worker02:/home/matej.hasul] ps -A -opid,uid,gid,comm,args | grep rabbit                                                             
  605   100   101 rabbitmq-server /bin/sh /opt/rabbitmq/sbin/rabbitmq-server                                                                     
  772   100     0 rabbitmq_export /rabbitmq_exporter
```

Reply:
Hmm, `runAsGroup` is not propagated to statefulset, however I can see it in generate template with `helm template`. Maybe issue with version? I'm using v1.12.3:
```
Client Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.3", GitCommit:"435f92c719f279a3a67808c80521ea17d5715c66", GitTreeState:"clean", BuildDate:"2018-11-26T12:57:14Z", GoVersion:"go1.10.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.3", GitCommit:"435f92c719f279a3a67808c80521ea17d5715c66", GitTreeState:"clean", BuildDate:"2018-11-26T12:46:57Z", GoVersion:"go1.10.4", Compiler:"gc", Platform:"linux/amd64"}
```

Reply:
if you do `kubectl get pod rabbitmq-0 -o yaml` does it show correct securityContext? Maybe a number to string conversion issue?

Reply:
I find `helm upgrade --debug --dry-run` output to be more useful for debugging issues.

Reply:
Thanks for hints, going to elaborate on this one. Output of suggested command is:
```
[root@ii-k8s-master01:~] kubectl get pod rabitmq-ha2-rabbitmq-ha-0 -o yaml | grep -A 5 security
  securityContext:
    runAsNonRoot: true
    runAsUser: 100
  serviceAccount: rabitmq-ha2-rabbitmq-ha
  serviceAccountName: rabitmq-ha2-rabbitmq-ha
  subdomain: rabitmq-ha2-rabbitmq-ha-discovery
```

Reply:
Output of `helm upgrade --debug --dry-run`:
```
COMPUTED VALUES:
.
.
.
securityContext:
  runAsGroup: 101
  runAsNonRoot: true
  runAsUser: 100
```
Generated statefulset:
```---
# Source: rabbitmq-ha/templates/statefulset.yaml
apiVersion: apps/v1beta1
kind: StatefulSet
metadata:
  name: rabitmq-ha2-rabbitmq-ha
  labels:
    app: rabbitmq-ha
    chart: rabbitmq-ha-1.21.1
    release: rabitmq-ha2
    heritage: Tiller
spec:
  podManagementPolicy: OrderedReady
  serviceName: rabitmq-ha2-rabbitmq-ha-discovery
  replicas: 3
  updateStrategy:
    type: OnDelete
  template:
    metadata:
      labels:
        app: rabbitmq-ha
        release: rabitmq-ha2
      annotations:
        checksum/config: 4038cd4a4d7be0d5aed52170b74bb0e5b3b01fe2df3226df95d5a220d77ee9d1
    spec:
      terminationGracePeriodSeconds: 10
      securityContext:
        runAsGroup: 101
        runAsNonRoot: true
        runAsUser: 100
```
I've been googling and it seems that `runAsGroup` is still in alpha stage sofar - https://github.com/kubernetes/enhancements/issues/213.

Reply:
yes planning to get it to beta in 1.14 k8s